manifests/mount.pp

```diff
@@ -1,34 +1,73 @@
+#   Check https://github.com/puppetlabs/puppetlabs-mount_core/blob/main/REFERENCE.md#atboot for more information.
+#
+# @param blockdevice
+#   Value passed to blockdevice attribute. Unused if empty.
+#   Check https://github.com/puppetlabs/puppetlabs-mount_core/blob/main/REFERENCE.md#blockdevice for more information.
+#
+# @param dump
+#   Value passed to dump attribute. Unused if empty.
+#   Check https://github.com/puppetlabs/puppetlabs-mount_core/blob/main/REFERENCE.md#dump for more information.
+#
+# @param options
+#   Value passed to options attribute. Unused if empty.
+#   Check https://github.com/puppetlabs/puppetlabs-mount_core/blob/main/REFERENCE.md#options for more information.
+#
+# @param pass
+#   Value passed to pass attribute. Unused if empty.
+#   Check https://github.com/puppetlabs/puppetlabs-mount_core/blob/main/REFERENCE.md#pass for more information.
+#
+# @param provider
+#   Value passed to provider attribute. Unused if empty.
+#   Check https://github.com/puppetlabs/puppetlabs-mount_core/blob/main/REFERENCE.md#provider for more information.
+#
+# @param remounts
+#   Value passed to remounts attribute. Unused if empty.
+#   Check https://github.com/puppetlabs/puppetlabs-mount_core/blob/main/REFERENCE.md#remounts for more information.
+#
+# @param target
+#   Value passed to target attribute. Unused if empty.
+#   Check https://github.com/puppetlabs/puppetlabs-mount_core/blob/main/REFERENCE.md#target for more information.
 #
 define types::mount (
-  $device,
-  $fstype,
-  $ensure      = mounted,
-  $atboot      = true,
-  $blockdevice = undef,
-  $dump        = undef,
-  $options     = undef,
-  $pass        = undef,
-  $provider    = undef,
-  $remounts    = undef,
-  $target      = undef,
+  String[1]                                                    $device,
+  String[1]                                                    $fstype,
+  Enum['defined', 'present', 'unmounted', 'absent', 'mounted'] $ensure      = 'mounted',
+  Boolean                                                      $atboot      = true,
+  Optional[String[1]]                                          $blockdevice = undef,
+  Optional[String[1]]                                          $dump        = undef,
+  Optional[String[1]]                                          $options     = undef,
+  Optional[String[1]]                                          $pass        = undef,
+  Optional[String[1]]                                          $provider    = undef,
+  Optional[Boolean]                                            $remounts    = undef,
+  Optional[String[1]]                                          $target      = undef,
 ) {
-
-  # validate params
-  validate_re($ensure, '^(present)|(unmounted)|(absent)|(mounted)$',
-    "types::mount::${name}::ensure is invalid and does not match the regex.")
-  validate_absolute_path($name)
-
   if $ensure != 'absent' {
     # ensure target exists
-    include ::common
+    include common
```

Comment:
Is this include really required? common::mkdir_p should be be accessible without the class being included in the catalogue.

Reply:
Agreed, even if common is a passive class it shouldn't be needed to be included.